### ISSUE TITLE
False positive from SC2184 (quote arguments to unset)

### ISSUE THREAD

**Issue by sig-daveka** [koalaman/shellcheck#2434]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2184
- My shellcheck version (`shellcheck --version` or "online"): 0.8.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
set -o noglob
unset data[1]

```

#### Here's what shellcheck currently says:

```
Line 3:
unset data[1]
      ^-- SC2184 (warning): Quote arguments to unset so they're not glob expanded.
```

#### Here's what I wanted or expected to see:

Nothing.  When `noglob` is set arguments are not glob expanded.  If it is too difficult to detect this you might mention it as an exception on https://github.com/koalaman/shellcheck/wiki/SC2184

**Comment by sig-daveka** [comment]
Sure, I had a go at updating the wiki to document this exception!
